tests/test_preload_script.py

```diff
@@ -489,6 +490,48 @@ async def test_addPreloadScriptGlobally_loadedInNewContexts(
     assert result["result"] == {"type": "string", "value": 'bar'}
 
 
+@pytest.mark.asyncio
+@pytest.mark.skip(reason="TODO: Implement test once selectors are supported.")
+async def test_addPreloadScriptGlobally_withBlankTargetLink(
+        websocket, context_id, create_context, html, type):
+    LINK_WITH_BLANK_TARGET = html(
+        '<a href="https://example.com" target="_blank">new tab</a>')
+
+    await execute_command(
+        websocket, {
+            "method": "script.addPreloadScript",
+            "params": {
+                "functionDeclaration": "() => { window.foo='bar'; }",
+            }
+        })
+
+    await execute_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": LINK_WITH_BLANK_TARGET,
+                "wait": "complete",
+                "context": context_id
+            }
+        })
+
+    result = await execute_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": "window.foo",
+                "target": {
+                    "context": context_id
+                },
+                "awaitPromise": True,
+                "resultOwnership": "root"
+            }
+        })
+    assert result["result"] == {"type": "string", "value": 'bar'}
+
+    # TODO: Click the link and assert that the new tab has the preload script.
```

Comment:
You can use BiDi+ to send [`Runtime.evaluate`](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#method-evaluate) with `userGesture: true`.

Reply:
```
    await send_JSON_command(
        websocket, {
            "method": "cdp.sendCommand",
            "params": {
                "cdpMethod": "Runtime.evaluate",
                "cdpParams": {
                    "expression": "document.querySelector('a').click();",
                    "userGesture": True
                },
                "cdpSession": cdp_session
            }
        })
    await read_JSON_message(websocket)
```

Is this what you mean?

Reply:
https://github.com/GoogleChromeLabs/chromium-bidi/pull/752/commits/4dc21c694f821cfe708b7912170f6da7b51a78fd

Reply:
Should we just add a TODO here for now and skip it as you previously suggested?

Reply:
if it does not work, you can skip test for now, and leave a todo

Reply:
did you try running it manually or in headful mode?

Reply:
https://github.com/GoogleChromeLabs/chromium-bidi/pull/752/commits/3c8d2c5e1b48760630ab3ea29fd7db12ea6eb696